source/draw/RasterRgb.ooc

```diff
@@ -0,0 +1,195 @@
+		result := This new(size)
+		result canvas draw(this, IntBox2D new(this size), IntBox2D new(size))
+		result
+	}
+	distance: func (other: Image) -> Float {
+		result := 0.0f
+		if (!other || (this size != other size))
+			result = Float maximumValue
+		else if (!other instanceOf(This)) {
+			converted := This convertFrom(other as RasterImage)
+			result = this distance(converted)
+			converted referenceCount decrease()
+		} else {
+			for (y in 0 .. this size y)
+				for (x in 0 .. this size x) {
+					c := this[x, y]
+					o := (other as This)[x, y]
+					if (c distance(o) > 0) {
+						maximum := o
+						minimum := o
+						for (otherY in 0 maximum(y - this distanceRadius) .. (y + 1 + this distanceRadius) minimum(this size y))
+							for (otherX in 0 maximum(x - this distanceRadius) .. (x + 1 + this distanceRadius) minimum(this size x))
+								if (otherX != x || otherY != y) {
+									pixel := (other as This)[otherX, otherY]
+									if (maximum blue < pixel blue)
+										maximum blue = pixel blue
+									else if (minimum blue > pixel blue)
+										minimum blue = pixel blue
+									if (maximum green < pixel green)
+										maximum green = pixel green
+									else if (minimum green > pixel green)
+										minimum green = pixel green
+									if (maximum red < pixel red)
+										maximum red = pixel red
+									else if (minimum red > pixel red)
+										minimum red = pixel red
+								}
+						distance := 0.0f
+						if (c blue < minimum blue)
+							distance += (minimum blue - c blue) as Float squared
+						else if (c blue > maximum blue)
+							distance += (c blue - maximum blue) as Float squared
+						if (c green < minimum green)
+							distance += (minimum green - c green) as Float squared
+						else if (c green > maximum green)
+							distance += (c green - maximum green) as Float squared
+						if (c red < minimum red)
+							distance += (minimum red - c red) as Float squared
+						else if (c red > maximum red)
+							distance += (c red - maximum red) as Float squared
+						result += (distance) sqrt() / 3
+					}
+				}
+			result /= this size length
+		}
+		result
+	}
+	swapRedBlue: func {
+		this swapChannels(0, 2)
+	}
+	redBlueSwapped: func -> This {
+		result := this copy()
+		result swapRedBlue()
+		result
+	}
+	_createCanvas: override func -> Canvas { RasterRgbCanvas new(this) }
+
+	operator [] (x, y: Int) -> ColorRgb { this isValidIn(x, y) ? ((this buffer pointer + y * this stride) as ColorRgb* + x)@ : ColorRgb new(0, 0, 0) }
+	operator []= (x, y: Int, value: ColorRgb) { ((this buffer pointer + y * this stride) as ColorRgb* + x)@ = value }
+
+	open: static func (filename: String) -> This {
+		x, y, imageComponents: Int
+		requiredComponents := 3
+		data := StbImage load(filename, x&, y&, imageComponents&, requiredComponents)
+		buffer := ByteBuffer new(data as Byte*, x * y * requiredComponents, true)
+		result := This new(buffer, IntVector2D new(x, y))
+		result swapRedBlue()
+		result
```

Comment:
You can of course add this function if you want, but this PR is quite big already.